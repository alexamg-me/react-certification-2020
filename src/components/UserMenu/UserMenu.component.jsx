import React, { useContext } from 'react';
import { useHistory } from 'react-router-dom';
import { Wrapper, Avatar } from './UserMenu.styled';
import { useAuth } from '../../providers/Auth';
import { GlobalContext } from '../../context/GlobalContext';

function UserMenu() {
  const history = useHistory();
  const { authenticated, logout } = useAuth();
  const globalContext = useContext(GlobalContext);

  function deAuthenticate(event) {
    event.preventDefault();
    globalContext.setSessionData({ loggedIn: false });
    logout();
    history.push('/');
  }
  const onClickHandler = (e) => {
    if (authenticated) {
      deAuthenticate(e);
    } else {
      history.push('/login');
    }
  };
  return (
    <Wrapper>
      <Avatar
        className={`${authenticated ? 'logged' : 'notlogged'}`}
        theme={globalContext.colors}
        onClick={onClickHandler}
      />
    </Wrapper>
  );
}

export default UserMenu;
