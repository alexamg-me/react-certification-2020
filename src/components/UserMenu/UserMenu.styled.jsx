import styled from 'styled-components';

const Wrapper = styled.div`
  display: inline-block;
`;
const Avatar = styled.div`
  display: block;
  height: 40px;
  width: 40px;
  border-radius: 100px;
  background-color: #409287;
  box-shadow: 0 0 10px #ffffff;
  cursor: pointer;
  &.logged {
    &:after {
      content: '🧑🏻';
      font-size: 35px;
      border-radius: 100px;
      text-align: center;
      width: 100%;
      display: block;
    }
  }
  &.notlogged {
    &:after {
      content: '👤';
      font-size: 35px;
      border-radius: 100px;
      text-align: center;
      width: 100%;
      display: block;
    }
  }
`;

export { Wrapper, Avatar };
