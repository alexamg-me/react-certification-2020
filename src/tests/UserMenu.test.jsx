import React from 'react';
import TestRenderer from 'react-test-renderer';
import UserMenu from '../components/UserMenu/UserMenu.component';

describe(`Component: UserMenu`, () => {
  test(`UserMenu renders with default props`, () => {
    const component = TestRenderer.create(<UserMenu />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
