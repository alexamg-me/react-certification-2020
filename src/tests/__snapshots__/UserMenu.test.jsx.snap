// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserMenu UserMenu renders with default props 1`] = `
<div
  className="sc-bdfBwQ bEhSYe"
>
  <div
    className="sc-gsTCUz fsihtp"
  />
</div>
`;
